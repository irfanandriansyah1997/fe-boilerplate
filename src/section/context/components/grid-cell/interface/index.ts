import { MouseEventHandler } from 'react';

/**
 * Grid Cell Props Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.10.31
 */
export interface IGridCellProps {
  cell: number;
  onClickCell: MouseEventHandler<HTMLButtonElement>;
}
