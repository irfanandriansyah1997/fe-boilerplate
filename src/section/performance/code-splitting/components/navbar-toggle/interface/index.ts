/**
 * Navbar Toggle Props Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.10.14
 */
export interface INavbarToggleProps {
  active: boolean;
  onToggle(active: boolean): void;
}
