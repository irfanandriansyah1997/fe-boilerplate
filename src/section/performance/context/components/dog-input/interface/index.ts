import { ChangeEventHandler } from 'react';

/**
 * Dog Input Props Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.10.31
 */
export interface IDogInputProps {
  dogName: string;
  onChangeDogName: ChangeEventHandler<HTMLInputElement>;
}
