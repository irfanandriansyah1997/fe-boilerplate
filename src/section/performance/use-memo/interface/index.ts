/**
 * City Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.10.28
 */
export interface ICity {
  country: string;
  id: string;
  lat: string;
  lng: string;
  name: string;
}

/**
 * City V2 Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.10.28
 */
export interface ICityV2 extends ICity {
  isActive?: boolean;
}
