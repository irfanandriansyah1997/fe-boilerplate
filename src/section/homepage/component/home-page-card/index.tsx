import { FC } from 'react';
import { NavLink } from 'react-router-dom';

import { IMenu } from '../../../../interface/component';
import styles from './style/style.module.css';

/**
 * Homepage Card Component
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.10.15
 */
const HomepageCard: FC<IMenu> = ({ menu, subtitle, title }) => (
  <div className={styles[`homepage-card`]}>
    <h6>{title}</h6>
    <div className={styles[`homepage-card__container`]}>
      <p>{subtitle}</p>
      <ul>
        {menu.map(({ text, to }) => (
          <li key={`${text}-${to}`}>
            <NavLink to={to}>
              <span className="material-icons">sticky_note_2</span>
              {text}
            </NavLink>
          </li>
        ))}
      </ul>
    </div>
  </div>
);

export default HomepageCard;
