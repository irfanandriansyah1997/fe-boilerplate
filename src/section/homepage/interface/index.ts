import { IMasonryItem } from '../../../components/organisms/masonry-layout/interface';

/**
 * App Menu List Item Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.11.17
 */
export interface IAppMenuListItem {
  dataTestID: string;
  item: IMasonryItem[];
  label: string;
}

/**
 * App Menu List Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.11.17
 */
export interface IAppMenuList {
  section: IAppMenuListItem[];
}
