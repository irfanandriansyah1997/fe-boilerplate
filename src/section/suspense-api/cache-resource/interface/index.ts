import { IResourcePayload } from '../../../../interface/general';
import { IPokemonDetail } from '../../../../interface/pokemon';

/**
 * Pokemon Cache Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.11.21
 */
export type IPokemonCache = (
  pokemonName: string
) => IResourcePayload<IPokemonDetail>;

/**
 * Pokemon Provider Props Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.11.21
 */
export interface IPokemonProviderProps {
  cacheTime?: number;
}
