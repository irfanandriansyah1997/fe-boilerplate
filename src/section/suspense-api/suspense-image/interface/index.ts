import { IResourcePayload } from '../../../../interface/general';
import { IPokemonDetail } from '../../../../interface/pokemon';

/**
 * Pokemon Cache Payload Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.11.21
 */
export interface IPokemonCachePayload {
  data: IResourcePayload<IPokemonDetail>;
  image: IResourcePayload<string>;
}

/**
 * Pokemon Cache Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.11.21
 */
export type IPokemonCache = (pokemonName: string) => IPokemonCachePayload;

/**
 * Pokemon Provider Props Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.11.21
 */
export interface IPokemonProviderProps {
  cacheTime?: number;
}
