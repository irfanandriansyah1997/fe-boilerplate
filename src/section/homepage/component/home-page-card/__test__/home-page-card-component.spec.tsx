import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { IMenuItem } from '../../../../../interface/component';
import HomepageCard from '..';

const MOCK_MENU: IMenuItem[] = [
  {
    text: `Wrap Use Memo`,
    to: `/context/part-1`
  },
  {
    text: `Separate Context Value`,
    to: `/context/part-2`
  },
  {
    text: `Colocate State`,
    to: `/context/part-3`
  }
];

describe(`Testing Homepage Card`, () => {
  it(`Snapshot Testing`, () => {
    const { getByTestId, queryByTestId } = render(
      <MemoryRouter>
        <HomepageCard
          menu={MOCK_MENU}
          subtitle="Sample Subtitle Course"
          title="Title Course"
        />
      </MemoryRouter>
    );

    expect(queryByTestId(/^homepage-card$/i)).toBeInTheDocument();
    expect(getByTestId(/^homepage-card$/i)).toMatchSnapshot();
  });

  it(`Should Render Title & Subtitle On Homepage Card`, () => {
    const { queryByText } = render(
      <MemoryRouter>
        <HomepageCard
          menu={MOCK_MENU}
          subtitle="Sample Subtitle Course"
          title="Title Course"
        />
      </MemoryRouter>
    );

    expect(queryByText(/^Title Course$/i)).toBeInTheDocument();
    expect(queryByText(/^Sample Subtitle Course$/i)).toBeInTheDocument();
  });

  it(`Should Render 3 Navlink On Homepage Card`, () => {
    const { queryAllByTestId } = render(
      <MemoryRouter>
        <HomepageCard
          menu={MOCK_MENU}
          subtitle="Sample Subtitle Course"
          title="Title Course"
        />
      </MemoryRouter>
    );

    expect(queryAllByTestId(/homepage-card-link/)).toHaveLength(3);
    const [firstElement, secondElement, thirdElement] = queryAllByTestId(
      /homepage-card-link/
    );

    /**
     * Check Attribute First Item
     */
    expect(firstElement.textContent).toBe(`sticky_note_2Wrap Use Memo`);
    expect(firstElement).toHaveAttribute(`href`, `/context/part-1`);

    /**
     * Check Attribute Second Item
     */
    expect(secondElement.textContent).toBe(
      `sticky_note_2Separate Context Value`
    );
    expect(secondElement).toHaveAttribute(`href`, `/context/part-2`);

    /**
     * Check Attribute Third Item
     */
    expect(thirdElement.textContent).toBe(`sticky_note_2Colocate State`);
    expect(thirdElement).toHaveAttribute(`href`, `/context/part-3`);
  });
});
