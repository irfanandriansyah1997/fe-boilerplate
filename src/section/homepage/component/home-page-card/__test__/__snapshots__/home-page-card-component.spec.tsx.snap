// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Homepage Card Snapshot Testing 1`] = `
<div
  class="homepage-card"
  data-testid="homepage-card"
>
  <h6>
    Title Course
  </h6>
  <div
    class="homepage-card__container"
  >
    <p>
      Sample Subtitle Course
    </p>
    <ul>
      <li>
        <a
          data-testid="homepage-card-link"
          href="/context/part-1"
        >
          <span
            class="material-icons"
          >
            sticky_note_2
          </span>
          Wrap Use Memo
        </a>
      </li>
      <li>
        <a
          data-testid="homepage-card-link"
          href="/context/part-2"
        >
          <span
            class="material-icons"
          >
            sticky_note_2
          </span>
          Separate Context Value
        </a>
      </li>
      <li>
        <a
          data-testid="homepage-card-link"
          href="/context/part-3"
        >
          <span
            class="material-icons"
          >
            sticky_note_2
          </span>
          Colocate State
        </a>
      </li>
    </ul>
  </div>
</div>
`;
